components/DonationAskModal.vue

```diff
@@ -66,106 +66,127 @@
+
+const targetMet = computed(() => {
+  return groupId.value && raised.value > target.value
+})
+
+function ask() {
+  show(props.requestedVariant)
+
+  miscStore.set({
+    key: 'lastdonationask',
+    value: new Date().getTime(),
+  })
+}
+
+const lastAsk = miscStore.get('lastdonationask')
+const canAsk =
+  !lastAsk || new Date().getTime() - lastAsk > 60 * 60 * 1000 * 24 * 7
+
+// Donation ask on Browse page is only used when we have a specific push.
+// canAsk = false // Why is this?
+
+if (canAsk) {
+  ask()
+}
+
+const { $bus } = useNuxtApp()
+$bus.$on('outcome', (params) => {
+  groupId.value = params.groupid
+  const { outcome } = params
+
+  if (outcome === 'Taken' || outcome === 'Received') {
+    // If someone has set up a regular donation, then we don't ask them to donate again.  Wouldn't be fair to
+    // pester them.
+
+    if (!me?.donorrecurring && canAsk) {
+      ask()
     }
-  },
-  computed: {
-    ...mapState(useDonationStore, ['target', 'raised']),
-    groupname() {
-      if (this.groupid && !this.targetMet) {
-        const group = this.groupStore?.get(this.groupid)
-
-        return group?.namedisplay
-      }
+  }
+})
 
-      return 'Freegle'
-    },
-    targetMet() {
-      // Have we met our per group target?  If so then we'll do a global donation appeal.
-      return this.groupid && this.raised > this.target
-    },
-  },
-  methods: {
-    async show(variant) {
-      // We need to decide which variant of donation ask to show.
-      this.variant = variant
-
-      try {
-        if (!variant) {
-          variant = 'buttons1'
-
-          variant = await this.api.bandit.choose({
-            uid: 'donation',
-          })
-
-          if (variant) {
-            this.variant = variant.variant
-          }
-        }
-      } catch (e) {
-        console.error('Get variant failed')
-      }
+async function show(requestedVariant) {
+  // We need to decide which variant of donation ask to show.
+  variant.value = requestedVariant
 
-      this.showModal = true
+  try {
+    if (!requestedVariant) {
+      requestedVariant = 'buttons1'
 
-      // Record the show
-      await this.api.bandit.shown({
-        uid: 'donation',
-        variant: this.variant,
-      })
-    },
-    score(value) {
-      this.api.bandit.chosen({
+      requestedVariant = await api.bandit.choose({
         uid: 'donation',
-        variant: this.variant,
-        score: value,
       })
-    },
-  },
+
+      if (requestedVariant) {
+        variant.value = requestedVariant.variant
+      }
+    }
+  } catch (e) {
+    console.error('Get variant failed')
+  }
+
+  showModal.value = true
+
+  // Record the show
+  await api.bandit.shown({
+    uid: 'donation',
+    variant: variant.value,
+  })
+}
+
+function score(value) {
+  api.bandit.chosen({
+    uid: 'donation',
+    variant: variant.value,
+    score: value,
+  })
 }
 </script>
+
+<style>
+#donationaskmodal {
+  opacity: 1;
+}
+</style>
```

Comment:
This I don't like, but without it the modal is always rendered with `opacity: 0`, which blocks it from being show.

Reply:
I'm not touching this one (discussed)